src/endpoint.jl

```diff
@@ -80,26 +89,39 @@ function Base.isequal(a::RightEndpoint, b::LeftEndpoint)
 end
 
 function Base.isless(a::LeftEndpoint, b::LeftEndpoint)
-    a.endpoint < b.endpoint || (a.endpoint == b.endpoint && a.included && !b.included)
+    return !(isunbounded(a.endpoint) && isunbounded(b.endpoint)) && (a.endpoint < b.endpoint || (a.endpoint == b.endpoint && a.included && !b.included))
 end
 
 function Base.isless(a::RightEndpoint, b::RightEndpoint)
-    a.endpoint < b.endpoint || (a.endpoint == b.endpoint && !a.included && b.included)
+    return !(isunbounded(a.endpoint) && isunbounded(b.endpoint)) && (a.endpoint < b.endpoint || (a.endpoint == b.endpoint && !a.included && b.included))
 end
 
 function Base.isless(a::LeftEndpoint, b::RightEndpoint)
-    a.endpoint < b.endpoint
+    return isunbounded(a.endpoint) || isunbounded(b.endpoint) || a.endpoint < b.endpoint
 end
 
 function Base.isless(a::RightEndpoint, b::LeftEndpoint)
-    a.endpoint < b.endpoint || (a.endpoint == b.endpoint && !(a.included && b.included))
+    return !isunbounded(a.endpoint) && !isunbounded(b.endpoint) && (a.endpoint < b.endpoint || (a.endpoint == b.endpoint && !(a.included && b.included)))
 end
 
+# Comparisons between Unbounded and Endpoints
+Base.isless(a::Endpoint{T}, b::Endpoint{T}) where T = a < b
+Base.isless(a::Endpoint{Nothing}, b::Endpoint{Nothing}) = false
+
+# Note: Are these two comparisons correct?
+Base.isless(a::Endpoint{Nothing}, b::Endpoint{T}) where T = true
+Base.isless(a::Endpoint{T}, b::Endpoint{Nothing}) where T = true
```

Comment:
I can't decide if this makes sense. Especially since we don't define a negative unbounded, and positive unbounded.